addons/l10n_id_efaktur/tests/test_l10n_id_efaktur.py

```diff
@@ -160,3 +161,43 @@ def test_efaktur_total_rounding_accuracy(self):
 
         self.assertEqual(amount_untaxed_total, amount_untaxed_sum)
         self.assertEqual(amount_tax_total, amount_tax_sum)
+
+    def test_efaktur_do_not_consume_code(self):
+        """ Ensure that an invoice with no taxes at all will not consume a code, while one with taxes will. """
+        available_code = self.efaktur.available
+        # 1. No taxes
+        out_invoice_no_taxes = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": False}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        })
+        out_invoice_no_taxes.action_post()
+        # The tax number is not set.
+        self.assertFalse(out_invoice_no_taxes.l10n_id_tax_number)
+        # No codes have been consumed.
+        self.assertEqual(self.efaktur.available, available_code)
+
+        with self.assertRaises(ValidationError, msg='E-faktur is not available for invoices without any taxes.'), self.cr.savepoint():
+            out_invoice_no_taxes.download_efaktur()
+
+        # 2 Taxes
```

Comment:
Since there is no setup in common, let's split into 2 different tests

Better to have smaller tests to ease the understanding, and to have a better overview of what is working or not